core/lib/dal/src/transactions_web3_dal.rs

```diff
@@ -24,171 +22,102 @@ pub struct TransactionsWeb3Dal<'a, 'c> {
 }
 
 impl TransactionsWeb3Dal<'_, '_> {
-    pub async fn get_transaction_receipt(
+    pub async fn get_transaction_receipts(
```

Comment:
Nit: May make sense to document that `hashes` are expected to be unique. It could be possible to signal using `hashes: &HashSet<H256>`; see also a comment regarding the output ordering.